### DISCUSSION TITLE
GNNExplainer  Report an error

### DISCUSSION THREAD

**Discussion by wy-st** [pyg-team/pytorch_geometric#4146] [Q&A]
Hello, I'm having a problem with GNNExplainer. 
```
        edge_index = distance_x.nonzero(as_tuple=False).t() # (N,N) ---(2, E)
        edge_weight = distance_x[edge_index[0], edge_index[1]]
        explainer = GNNExplainer(net, epochs=0, return_type='log_prob')
        node_idx = 10
        node_feat_mask, edge_mask = explainer.explain_node(node_idx=node_idx, x=torch.from_numpy(prediction[0, :, 0]), edge_index=edge_index)
        ax, G = explainer.visualize_subgraph(node_idx, edge_index, edge_mask, y=torch.from_numpy(data_target_tensor[0, :, 0]))
        plt.show()
```
I modified the code according to your sample and used it in my experiment. But I send him always report the error as follows:
```
Traceback (most recent call last):
  File "D:\GNNExplain\lib\utils.py", line 363, in predict_and_save_results_mstgcn
    node_feat_mask, edge_mask = explainer.explain_node(node_idx=node_idx, x=torch.from_numpy(prediction[0, :, 0]), edge_index=edge_index)
  File "D:\anaconda\lib\site-packages\torch_geometric\nn\models\gnn_explainer.py", line 173, in explain_node
    out = self.model(x=x, edge_index=edge_index, **kwargs)
  File "D:\anaconda\lib\site-packages\torch\nn\modules\module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
TypeError: forward() got an unexpected keyword argument 'edge_index'

Process finished with exit code 1
```

**Comment by rusty1s** [comment]
How does your underyling model look like? Does it expect `edge_index` as `forward` argument?

**Comment by wy-st** [comment]
I am very sorry that I could not reply in time due to the graduation thesis and other related contents.
The model I use is a GCN paired with a GRU model (as known as T-GCN). The input dimensions of the data are, (Batch, Num_nodes, Feature, Time_stamp). In the GCN run, I put Batch together with Num_nodes. At the same time, the adjacency matrix I have also done to number the Node's in the form of Batch. But GNNExplainer cannot use.

**Reply by rusty1s** [reply]
What do you mean that `GNNExplainer` cannot use it? As long as graphs are properly batched according to the PyG format (stacking edge indices diagonally), there hopefully shouldn't be any problem. In general though, there does not exist a real need to batch graphs together for explaining them. This can also be done for each graph individually.

**Reply by wy-st** [reply]
For my major, I hope to explain the migration behavior at spatial nodes in terms of temporal evolution. Static graphs are used in the T-GCN model, and in practice, I hope to use dynamic graphs. Thus, I I turned the edge of the graph into a diagonal matrix.
